src/std_extensions/arithmetic/int_types.rs

```diff
@@ -15,113 +14,103 @@ use crate::{
 /// The extension identifier.
 pub const EXTENSION_ID: SmolStr = SmolStr::new_inline("arithmetic.int.types");
 
-/// Identfier for the integer type.
+/// Identifier for the integer type.
 const INT_TYPE_ID: SmolStr = SmolStr::new_inline("int");
 
-fn int_custom_type(n: u8) -> CustomType {
-    CustomType::new(
-        INT_TYPE_ID,
-        [TypeArg::USize(n as u64)],
-        EXTENSION_ID,
-        TypeBound::Copyable,
-    )
+fn int_custom_type(width_arg: TypeArg) -> CustomType {
+    CustomType::new(INT_TYPE_ID, [width_arg], EXTENSION_ID, TypeBound::Copyable)
 }
 
 /// Integer type of a given bit width.
 /// Depending on the operation, the semantic interpretation may be unsigned integer, signed integer
 /// or bit string.
-pub fn int_type(n: u8) -> Type {
-    Type::new_extension(int_custom_type(n))
+pub fn int_type(width_arg: TypeArg) -> Type {
+    Type::new_extension(int_custom_type(width_arg))
```

Comment:
Given this function is `pub` (should it be??), you should probably do some validation here. Or are we leaving that to when we have the extension registry available in `Hugr::validate`?

Reply:
So lots of things:

- The function was only used by the operation definitions, which should have their arguments validated also, so I have just made this method pub(super).
- However, the ops all listed their type params as TypeParam::max_usize(), which is incorrect! Now they all list the correct bounded parameter, so the check should happen. 
- Noting that direct access to valid integer types might be useful, I've added a lazy static array of the valid types. So all pub access should be correctly validated.


All that is to say...I wish we had more tests of this.